### ISSUE TITLE
Problems with T3 read().value compatibility in T4

### ISSUE THREAD

**Issue by malvarezcells** [taurus-org/taurus#757]
Hi all,

The taurus 3 compatibility layer in taurus 4 has a bug when the taurus attribute object is created and read using the old instruction in the same line. See the following example in ipython:
```
===================================================
In [1]: import taurus

In [2]: taurus.Release.version
Out[2]: '4.3.1-alpha'

In [3]: value = taurus.Attribute('sys/tg_test/1/float_scalar').read().value
MainThread     WARNING  2018-05-31 00:56:47,926 TaurusRootLogger: <ipython-input-3-2e16a4fdc439>:1: DeprecationWarning: _get_value is deprecated since 4.0. Use .rvalue instead
  value = taurus.Attribute('sys/tg_test/1/float_scalar').read().value

---------------------------------------------------------------------------
ReferenceError                            Traceback (most recent call last)
<ipython-input-3-2e16a4fdc439> in <module>()
----> 1 value = taurus.Attribute('sys/tg_test/1/float_scalar').read().value

/usr/lib/python2.7/dist-packages/taurus/core/util/log.pyc in new_func(*args, **kwargs)
   1053     def new_func(*args, **kwargs):
   1054         deprecated(dep=func.__name__, alt=alt, rel=rel, dbg_msg=dbg_msg)
-> 1055         return func(*args, **kwargs)
   1056 
   1057     doc = (func.__doc__ or '')

/usr/lib/python2.7/dist-packages/taurus/core/tango/tangoattribute.pyc in _get_value(self)
    155         """for backwards compat with taurus < 4"""
    156         try:
--> 157             return self.__fix_int(self.rvalue.magnitude)
    158         except AttributeError:
    159             return self.rvalue

/usr/lib/python2.7/dist-packages/taurus/core/tango/tangoattribute.pyc in __fix_int(self, value)
    217         Works on scalar and non-scalar values
    218         """
--> 219         if self._attrRef.type is None or self._attrRef.type != DataType.Integer:
    220             return value
    221         try:

ReferenceError: weakly-referenced object no longer exists

In [4]:
===============================================

But if the taurus attribute object is first referenced into a variable and the value is read in the following line, it works (despite the warning message encouraging to use the new method read().rvalue.magnitude). See the following example in a new ipython console:

=========================================
In [1]: import taurus

In [2]: taurus.Release.version
Out[2]: '4.3.1-alpha'

In [3]: ta = taurus.Attribute('sys/tg_test/1/float_scalar')

In [4]: ta.read().value
MainThread     WARNING  2018-05-31 01:03:59,562 TaurusRootLogger: <ipython-input-4-23a3e4973131>:1: DeprecationWarning: _get_value is deprecated since 4.0. Use .rvalue instead
  ta.read().value

Out[4]: 1843.0
===============================================

BTW, this bug is not there in taurus 3.

=============================================
In [1]: import taurus

In [2]: taurus.Release.version
Out[2]: '3.7.2'

In [3]: value = taurus.Attribute('sys/tg_test/1/float_scalar').read().value

In [4]: value
Out[4]: 1850.0
==============================================
```
Best regards,
Marc

**Comment by cmft** [comment]
Well done!! I confirm the bug. Many thanks!!!

**Comment by cpascual** [comment]
I am working on a fix.

**Comment by cpascual** [comment]
Hi, I think the bug is fixed in #758
